### DISCUSSION TITLE
[NEW] CC2674 (CC2674P/CC2674R) and CC1354 (CC1354P) Zigbee Coordinator adapters + firmware?

### DISCUSSION THREAD

**Discussion by Hedda** [Koenkk/Z-Stack-firmware#476] [General]
Has anyone yet started to look at CC2674P or CC2674R (CC2674x10) Zigbee Coordinator and Zigbee Router adapters and firmware?

https://www.ti.com/product/CC2674P10

https://www.ti.com/product/CC2674R10

https://www.ti.com/document-viewer/CC2674P10/datasheet

https://www.ti.com/document-viewer/CC2674R10/datasheet

https://www.ti.com/document-viewer/lit/html/swra765

https://www.ti.com/lit/gpn/CC2674P10

https://www.ti.com/lit/gpn/cc2674r10

Developers should check out the official "LP-EM-CC1354P10" (CC1354P10 LaunchPad) development kits hardware from Texas Instruments:

* https://www.ti.com/product/CC1354P10#hardware-development
  * https://www.ti.com/tool/LP-EM-CC1354P10
    * https://www.ti.com/tool/LP-XDS110
  * https://www.ti.com/tool/LP-XDS110ET

The new CC2674 features up to 1MB flash and 296KB RAM compared to CC2652 which can have up to 704KB flash and 152KB RAM at most.

Understand CC2674P10 should be able to be a replacement of CC2652P2/CC2652P7 with both coming in the same package size and format?

https://www.ti.com/wireless-connectivity/zigbee/products.html#1241=Wireless%20MCU&

https://www.ti.com/wireless-connectivity/zigbee/overview.html

![image](https://github.com/Koenkk/Z-Stack-firmware/assets/6320001/2f2064d1-7c36-47b8-a0e3-a7c9057c4a86)

![image](https://github.com/Koenkk/Z-Stack-firmware/assets/6320001/2a954f39-2c68-4568-9839-aa69fc370623)

![image](https://github.com/Koenkk/Z-Stack-firmware/assets/6320001/358e1b98-621a-4ca2-8616-19f176305eb5)

![image](https://github.com/Koenkk/Z-Stack-firmware/assets/6320001/fadfed27-c124-4b67-b8ea-1eae2c61bc1b)

**Comment by Hedda** [comment]
Missed that Texas Instruments have also released a CC1354P (CC1354P10) Sub-1 GHz and 2.4 GHz SoC with multi-band/dual-band support:

https://www.ti.com/product/CC1354P10

https://www.ti.com/document-viewer/cc1354p10/datasheet

https://www.ti.com/lit/gpn/cc1354p10

They have LP-EM-CC1354P10 as a LaunchPad Development Kit which looks to be available from only $29.00 (USD):

https://www.ti.com/tool/LP-EM-CC1354P10

![image](https://github.com/Koenkk/Z-Stack-firmware/assets/6320001/e61756ab-9b6b-4c09-a97b-42fd72e2f611)

In addition, they got info on how to create a 433 to 930-MHz and 2.4-GHz Tunable PCB Antenna

https://www.ti.com/lit/pdf/swra730

https://www.ti.com/document-viewer/lit/html/swra730

PS: Off-topic but find CC1354P with Sub-GHz really interesting as Zigbee Sub-GHz (sub-1 GHz) for smart home automation use is a long time coming (though the technology has already existed for many years and is today commonly used for utilities Zigbee Smart Energy (ZSE / Zigbee SE) metering in many commercial meters. The biggest thing missing has been multi-band/dual-band and dual-mode support in a single chip that is as inexpensive as the others.

https://csa-iot.org/newsroom/subghz-reliable-mesh/

https://rethinkresearch.biz/articles/zigbee-gets-dual-mode-2-4ghz-sub-ghz-spec-longer-range-mesh/

**Reply by LordNex** [reply]
Now I'm going to end up having to buy a new Adapter lol. Actually I'm hoping to mod my Google Home V2s in Onju and bring them into HA as private devices. They have the preprinted boards so I've been working with some people to get a large order for the US. But it looks like the Chrystal's in it are causing the price rise due to chip shortage. We're hoping to get the finished drop in ESP built boards for under $30 USD + shipping

**Reply by Hedda** [reply]
@LordNex  Nice but that is off-topic here.

**Comment by Hedda** [comment]
It looks like "CC1354P10-1 LaunchPad" and "CC1354P10-6 LaunchPad" is supported in SimpleLink CC13xx CC26xx SDK version 7.10.01.24?

I have not yet seen any news about a CC2674P10 or CC2674R10 based LaunchPad, but Mouser and others now look to have SoCs in stock.

* CC2674P106T0RGZR (CC2674P10 SoC)
* CC2674R106T0RGZR (CC2674R10 SoC)

@Koenkk Do you know if CC26x4 and CC13x4 will be supported by the same SimpleLink SDK and Z-Stack_3.x.0 as CC13x2 and CC26x2?

https://www.ti.com/tool/download/SIMPLELINK-LOWPOWER-F2-SDK

https://software-dl.ti.com/simplelink/esd/simplelink_cc13xx_cc26xx_sdk/7.10.01.24/exports/release_notes_simplelink_cc13xx_cc26xx_sdk_7_10_01_24.html

https://dev.ti.com/tirex/explore/node?node=A__ABWp1w.mXU7n8HjfEIyLGQ__com.ti.SIMPLELINK_CC13XX_CC26XX_SDK__BSEc4rl__LATEST

**Reply by Koenkk** [reply]
It requires a new firmware, znp projects from the new SDK:

<img width="372" alt="Screenshot 2023-10-24 at 19 53 38" src="https://github.com/Koenkk/Z-Stack-firmware/assets/2892853/d29bab78-4352-4f80-b6f4-5355f3eb6a02">

**Reply by Hedda** [reply]
Is anyone in the community looking at making CC1354P and/or CC2674P USB dongles(s) as Zigbee Coordinator hardware adapter(s)?

I am guessing that initially CC1354 and CC2674 USB adapters with easily accessed debug port on the board would be of most interest so that developers and testers start experiment with these new radio chips.

**Reply by Koenkk** [reply]
Not that I know off

**Reply by mercenaruss** [reply]
Why not, if chips are available i will try to make it or will request for modules from RF-STAR on this chip.

**Reply by Hedda** [reply]
@mercenaruss any news from RFstar on CC2674P and/or CC1354P radio modules?

Also check out post below from @LSagiroglu about E72-2G4M20S1C (CC2674P10) radio module from Ebyte/Cdebyte.

**Reply by mercenaruss** [reply]
@Hedda I will order today and make some sample. If @Koenkk need one i will send to him also.
You are interested in one sample to speed up firmware development?

**Reply by Hedda** [reply]
> You are interested in one sample to speed up firmware development?

Maybe once you and Koenkk get some CC2674P USB radio adapter prototypes and Z-Stack NCP Zigbee Coordinator firmware working with zigbee-herdsman for Zigbee2MQTT (and IoBroker) then maybe also ask if puddly as the developer of the zigpy-znp radio library want a prototype for sample too for broader testing (since zigpy is used by Home Assistant’s ZHA and the Zigbee Plugin for Domoticz) -> https://github.com/zigpy/zigpy-znp

**Reply by mercenaruss** [reply]
perfect,thanks.
I make the order and already prepared the PCB

**Comment by LSagiroglu** [comment]
module is ready: [https://www.cdebyte.com/products/E72-2G4M20S1C](https://www.cdebyte.com/products/E72-2G4M20S1C) (CC2674), pin to pin compatible with [https://www.cdebyte.com/products/E72-2G4M20S1E](https://www.cdebyte.com/products/E72-2G4M20S1E) (CC2652P)

**Reply by mercenaruss** [reply]
Will order today and start development.

**Reply by Hedda** [reply]
@LSagiroglu off-topic but wondering if Ebyte is also working on EFR32MG24 based radio modules as well?

Silicon Labs EFR32MG24 is the direct competition to Texas Instruments CC2674P and RFStar has released a such module:

https://www.rfstariot.com/rf-bm-mg24b2-matter-zigbee-openthread-ble-multiprotocol-efr32mg24-module_p106.html

https://www.rfstariot.com/rf-bm-mg24b1-ble5-3-matter-zigbee-thread-efr32mg24-module_p105.html

https://www.silabs.com/wireless/zigbee/efr32mg24-series-2-socs

Silicon Labs themselves have released its own EFR32MG24 based reference module as MGM240 (MGM240P and MGM240L):

https://www.silabs.com/wireless/zigbee/efr32mg24-series-2-modules

**Reply by codmpm** [reply]
Same here. Already ordered some CC2674 from ebyte. No word on the IPEX version, yet. Waiting for a reply from my sales representative.

**Reply by KrzysztofHajdamowicz** [reply]
@mercenaruss have your modules arrived?

**Comment by Hedda** [comment]
FYI, SMLIGHT has announced/launched CC2674P (CC2674P10) based "SLZB-07p10" USB dongle and "SLZB-06p10" Ethernet/WiFi adapter:

SLZB-07p10 -> https://smlight.tech/product/slzb-07p10/

SLZB-06p10 -> https://smlight.tech/product/slzb-06p10/

It is unclear if these are shipping already, and there are no Zigbee (or Thread/OpenThread) firmware images shared in their GitHub repos yet:

https://github.com/smlight-dev

Their website says it is supported by Zigbee2MQTT but believe that might just because they copies & pasted their webpages for their other models without updating anything but the model name, or at least noted those webpages still points the older user manual for some other models.

Maybe @smlight-dev-223942 / @Tarik2142 are to tell you/us more about the firmware build config that ship pre-flashed on these and if they are already shipping or not?

PS: I first spotted announced about these on CNXSoft's blog and that mentions that p10 models are only suitable for developers right now since the Zigbee firmware is still under development, see -> https://www.cnx-software.com/2024/02/13/zigbee-ethernet-wifi-coordinators-usb-adapters-ti-cc2652p7-cc2674p10-wireless-chips/

**Reply by Hedda** [reply]
> FYI, SMLIGHT has announced/launched CC2674P (CC2674P10) based "SLZB-07p10" USB dongle and "SLZB-06p10" Ethernet/WiFi adapter:
> 
> SLZB-07p10 -> https://smlight.tech/product/slzb-07p10/
> 
> SLZB-06p10 -> https://smlight.tech/product/slzb-06p10/

@Koenkk have you already got samples of this CC2674P10 based hardware and/or have you looked at making firmware builds for it?

```
ZIGBEE FIRMWARE FOR ZIGBEE CHIP CC2674P10 IS STILL UNDER DEVELOPMENT.

YOU CAN BUY THIS DEVICE ONLY FOR DEVELOPMENT PURPOSES

CC2674P10 GPIOS for developers (corresponds to LAUNCHXL board CC2652P):

RF: RF_24GHZ → DIO28, RF_HIGH_PA → DIO29;
UART: TX → DIO13, RX → DIO12;
LEDs: GREEN LED → DIO6, RED LED → DIO7;
BOOT: BSL → DIO15, Reset → RESET_N;
```

**Reply by Koenkk** [reply]
Yes, but the FW is not working yet (multiple people are working on this)

**Reply by Hedda** [reply]
Any updates on Z-Stack Zigbee Coordinator firmware (and/or Zigbee Router firmware) for CC2674/CC2674P or CC1354/CC1354R?

Noticed that SMLIGHT is now shipping their CC2674P1 based SLZB-07p10 USB adaper dongle:

https://smlight.tech/product/slzb-07p10/

Note! They updated that webpage to specifically state there that only for developers for now "_**ZIGBEE FIRMWARE FOR ZIGBEE CHIP CC2674P10 IS STILL UNDER DEVELOPMENT**_" and that "_**YOU CAN BUY THIS DEVICE ONLY FOR DEVELOPMENT PURPOSES**_".

CC2674P10 GPIOS for developers (corresponds to LAUNCHXL board CC2652P):
* RF: RF_24GHZ → DIO28, RF_HIGH_PA → DIO29;
* UART: TX → DIO13, RX → DIO12;
* LEDs: GREEN LED → DIO6, RED LED → DIO7;
* BOOT: BSL → DIO15, Reset → RESET_N;

**Reply by Hedda** [reply]
> Noticed that SMLIGHT is now shipping their CC2674P1 based SLZB-07p10 USB adaper dongle:
> 
> https://smlight.tech/product/slzb-07p10/
> 
> Note! They updated that webpage to specifically state there that only for developers for now "_**ZIGBEE FIRMWARE FOR ZIGBEE CHIP CC2674P10 IS STILL UNDER DEVELOPMENT**_" and that "_**YOU CAN BUY THIS DEVICE ONLY FOR DEVELOPMENT PURPOSES**_".
> 

@Koenkk Any updates on Z-Stack Zigbee Coordinator (and/or Zigbee Router firmware) for the new CC2674P (CC2674) chip?

Noticed SMLIGHT has now removed their webpage comment about CC2674P based SLZB-07p10 being only for development purposes:

* https://smlight.tech/product/slzb-07p10/

Also noticed SMLIGHT started selling that CC2674P based SLZB-07p10 wholesale on their store for $19.99(US) but says no stock now:

* https://smartlight.me/smart-home-devices/zigbee-devices/coordinator-slzb-07p10

As well noticed you officially released build of Z-Stack_3.x.0 coordinator 20240710 firmware version as new “master” for cc26x2/cc13x2:

* https://github.com/Koenkk/Z-Stack-firmware/discussions/505

  * https://github.com/Koenkk/Z-Stack-firmware/releases/tag/Z-Stack_3.x.0_coordinator_20240710

Wondering if people working on CC2674P firmware looking at SimpleLink SDK 7.41.00.17 as the base for Z-Stack coordinator FW too, or?

* https://software-dl.ti.com/simplelink/esd/simplelink_cc13xx_cc26xx_sdk/7.41.00.17/exports/changelog.html

PS: I see SimpleLink SDK 7.41.00.17 version mention "cc13x4" (CC1354R) and "cc26x4", (though not CC2674P or CC2674R specifically).

**Reply by Koenkk** [reply]
> Any updates on Z-Stack Zigbee Coordinator (and/or Zigbee Router firmware) for the new CC2674P (CC2674) chip?

I did not start looking into this yet (but already have an adapter with this chip)

**Comment by Hedda** [comment]
Off-topic for Z-Stack Zigbee firmware discussion but FYI, hardware developers might be interested that these now have support in Matter:

https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/ti

There are a few sections there that might still be related here too as they mention that GPIO pin values will require modifications, check out:

https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/ti/matter_cc2674_migration.md

# Running Matter Examples on the TI SimpleLink CC2674P10 and CC2674R10

The existing Matter project examples are based on LP_EM_CC1354P10_6. If using
the CC2674P10, the following migration steps are required. Developers are
strongly encouraged to start with a `cc13x4_26x4` example and migrate the
project accordingly. Example projects can be found in the following location:
`matter/examples/[application]/cc13x4_26x4`

## Dependencies

The following must be installed on your system before proceeding:

-   [SysConfig](https://www.ti.com/tool/SYSCONFIG) v1.16.2 or later

## Matter source code changes

The following are changes to the Matter source code files which should be
applied to convert a `matter/examples/[application]/cc13x4_26x4` project to the
CC2674P10 device

-   `examples/[application]/cc13x4_26x4/args.gni`, modify/add the following
    defines for the CC2674
    -   `ti_simplelink_board = "CC2674"`
    -   `ti_simplelink_device = "CC2674P10RGZ"`
-   `third_party/ti_simplelink_sdk/repo_cc13xx_cc26xx/source/ti/common/flash/no_rtos/extFlash/bsp.h`,
    modify the SPI GPIO pins to the value below:

```
#define BSP_IOID_FLASH_CS       IOID_20
#define BSP_SPI_MOSI            IOID_9
#define BSP_SPI_MISO            IOID_8
#define BSP_SPI_CLK_FLASH       IOID_10

```

The GPIO pin values for SPI will need to be adjusted based on your design.

## Configuring `chip.syscfg` in the SysConfig GUI

1. To open `matter/examples/[application]/cc13x4_26x4/chip.syscfg` in the GUI,
   add the following line to the top of the file:

```
  // @cliArgs --product <path to SDK>/.metadata/product.json --board /ti/boards/LP_EM_CC1354P10_6 --rtos freertos
```

2. Open the `syscfg` file using the standalone Sysconfig GUI
   (`sysconfig_gui.sh`) from the SysConfig installation folder.
3. Click on _Show Device View_ and then click _Switch_.
4. Select _Board_ as _None_ and _Device_ as _`CC2674P10RGZ`_, Unselect
   _`Lock PinMux`_, and click _Confirm_.
5. To fix errors, make the following module changes:
    - _RF Design_ and _RF Stacks -> BLE -> Radio_: click on _accepting the
      current value_, which should be _`LP_CC2674P10_RGZ`_ in the drop down menu
      for _Based On RF Design_
    - _TI DEVICES -> Device Configuration_: Clear
      _`XOSC Cap Array Modification`_
    - _TI DRIVERS -> RF_: Set _Global Event Mask_ as _None_ and _No. of Antenna
      Switch Control Pins_ as _0_
    - _TI DRIVERS -> UART2 -> `PinMux`_: Set _UART Peripheral_ to _UART0_, _TX
      Pin_ to _`DIO_13/19`_, and _RX Pin_ to _`DIO_12/18`_
    - _TI DRIVERS APPS -> Button_: Set _`PinMux`_ of _CONFIG_BTN_LEFT_ to
      _`DIO_15`_ and _CONFIG_BTN_RIGHT_ to _`DIO_14`_
    - _TI DRIVERS APPS -> LED_: Set _`PinMux`_ of _CONFIG_LED_RED_ to _`DIO_6`_
      and _CONFIG_LED_RIGHT_ to _`DIO_7`_
6. Save the SysConfig file (click on _Save As_) and ensure the file name matches
   the reference from `BUILD.gn` (default project name is `chip.syscfg`).
7. Open the new SysConfig file with a text editor and remove the generated
   arguments.
    ```
    /**
     * These arguments were used when this file was generated.  They will be automatically applied on subsequent loads
     * via the GUI or CLI.  Run CLI with '--help' for additional information on how to override these arguments.
     * @cliArgs --device "CC2674P10RGZ" --package "RGZ" --part "Default" --rtos "freertos" --product "simplelink_cc13xx_cc26xx_sdk@7.10.01.24"
     * @versions {"tool":"1.18.0+3130"}
     */
    ```
8. Move the `*.syscfg` file into the
   `matter/examples/[application]/cc13x4_26x4/` folder. Make sure that the
   `args.gni` parameters are aligned for the `CC2674P10RGZ` as detailed above,
   and build the example using the `README.md` instructions.

## Modifications required for the CC2674R10

After applying all items in the "Configuring `chip.syscfg` in the SysConfig GUI"
section, additional steps must also be applied to generate Matter project for
the CC2674R10.

-   `examples/[application]/cc13x4_26x4/args.gni` should have
    `ti_simplelink_board` as `CC2674` and `ti_simplelink_device = CC2674R10RGZ`.
-   `examples/[application]/cc13x4_26x4/chip.syscfg` opened with a Text Editor
    should change `ble.radioConfig.codeExportConfig.$name` to
    `ti_devices_radioconfig_code_export_param2` and `ble.rfDesign` to
    `LP_EM_CC1354P10_1`

Furthermore, the subsequent changes apply specifically for the CC2674R10 and
should be addressed from a SysConfig Editor.

1. Pins will need to be reconfigured as such:

    | SysConfig pin name | R10 `PinMux` |
    | ------------------ | ------------ |
    | UART_RX            | `DIO_2`      |
    | UART_TX            | `DIO_3`      |
    | CONFIG_BTN_LEFT    | `DIO_13`     |
    | CONFIG_BTN_RIGHT   | `DIO_14`     |
    | CONFIG_LED_RED     | `DIO_6`      |
    | CONFIG_LED_GREEN   | `DIO_7`      |

2. _Custom -> IEEE 802.15.4-2006, `250 kbps`, `OQPSK`, `DSSS = 1:8` -> Code
   Export Configuration_, acknowledge and dismiss the PA radio setup error
3. _Custom -> IEEE 802.15.4-2006, `250 kbps`, `OQPSK`, `DSSS = 1:8` -> RF
   Command Symbols_, change `CMD_RADIO_SETUP` from `RF_cmdRadioSetup` to
   `RF_cmdIeeeRadioSetup` and add the following functions from the drop-down:
   `CMD_TX_TEST`,`CMD_IEEE_ED_SCAN`, `CMD_IEEE_CSMA`, and `CMD_IEEE_RX_ACK`.

## Building examples for the CC1354P10-1

To migrate the CC1354P10-6 examples to the CC1354P10-1 platform, there are only
two steps:

1. `examples/[application]/cc13x4_26x4/args.gni` should have
   `ti_simplelink_board` as `CC1354P10-1`
2. `examples/[application]/cc13x4_26x4/chip.syscfg` opened with a Text Editor
   should change `ble.radioConfig.codeExportConfig.$name` to
   `ti_devices_radioconfig_code_export_param2` and `ble.rfDesign` to
   `LP_EM_CC1354P10_1`

After this, the example's `README.md` instructions can be followed to produce
the executable needed.

**Comment by Hedda** [comment]
FYI, JelmerT's upstream [cc2538-bsl](https://github.com/JelmerT/cc2538-bsl/) (programming/flash command-line tool) does not yet support CC2674 or CC1354 but Texas Instruments  looks to include a forked variant of it which someone wrote supposedly should support flashing the newer CC2674 and CC1354 SoCs, see:

- https://github.com/TexasInstruments/simplelink-lowpower-f2-sdk/tree/main/source/third_party/cc2538-bsl

It would be appriciated if someone could test TI's forked variant to confirm compatibility with CC2674/CC1354 and push patches upstream:

* https://github.com/JelmerT/cc2538-bsl/issues/160

**Reply by Hedda** [reply]
> FYI, JelmerT's upstream [cc2538-bsl](https://github.com/JelmerT/cc2538-bsl/) (programming/flash command-line tool) does not yet support CC2674 or CC1354 but Texas Instruments looks to include a forked variant of it which someone wrote supposedly should support flashing the newer CC2674 and CC1354 SoCs, see:
> 
> * https://github.com/TexasInstruments/simplelink-lowpower-f2-sdk/tree/main/source/third_party/cc2538-bsl

FYI, the SMLIGHT team has now also released a new web flasher on a cc2538-bsl with added support for CC2674P P10 (CC2674P10):

* https://github.com/smlight-tech/smlight-cc-web-tools

That webflasher depends on their "smlight_cc_flasher" project which in turn is another fork of JelmerT's cc2538-bsl project:

* https://github.com/smlight-tech/smlight-cc-flasher

*  https://pypi.org/project/smlight-cc-flasher/

darkxst wrote that the smlight-cc-flasher is based on a heavily modified + improved experimental fork of JelmerT's cc2538-bsl project and that tthe fork has been tested with the new [SMLIGHT SLZB-07p10](https://smlight.tech/product/slzb-07p10/) Zigbee Coordinator USB adapter based in CC2674P P10 (CC2674P10) that they released a while ago (but still mark as sold for developers only as there is no well tested community firmware publicly available as of yet):

* https://smlight.tech/flasher/#SLZB-07p10

Again, please also see related issue tracker requesting support for CC2674 (CC2674P/CC2674R) and CC1354 (CC1354P) compatibility:

* https://github.com/JelmerT/cc2538-bsl/issues/160

**Comment by holsteiner-kiel** [comment]
Hi. What's up with Support for CC2474? I have an Smlight SLZB06-P10. Think ITS out for months and not supported yet. Is it correct?

Thank you very much!

**Comment by lux73** [comment]
just wanted to let you know -> this [CC2674P10](https://www.domadoo.fr/en/dongle-zigbee/7611-smlight-zigbee-cc2674p10-soc-usb-dongle-3db-antenna-zigbee2mqtt.html) is shipped with already flashed FW20240705 and works OotB with z2m 2.1.1

![z2m2 1 1](https://github.com/user-attachments/assets/a151adf5-6686-4c6c-bf49-e6ec975cc238)

![z2m_cc2674p10](https://github.com/user-attachments/assets/795ce77b-2f30-416d-8611-3736f63ab1b3)

i'll be able to pair new Devices - I'll test for a few Weeks and then replace my CC2652RB (which also works stably)

**Reply by Hedda** [reply]
That is the same SMLIGHT SLZB-07p10 which is mentioned and linked several times above. Search for ”SLZB-07p10” on this page. Still only experimental and as you may notice from the date on the firmware (revision 2024-07-05) there is no more updates on firmware development for stable/mature firmware for non-developers.

**Comment by Koenkk** [comment]
A CC2674P10 fw is now available; https://github.com/Koenkk/Z-Stack-firmware/discussions/545

**Reply by eiSouthBoy** [reply]
Hi @Koenkk :

> CC2674P10_coordinator_20250321.hex.zip: not available for now, wip

I'd like to know if the firmware is stable and usable, and if there are any plans to release the firmware file?

**Reply by KimCarlsen** [reply]
hi @Koenkk any update on the release build of this?

**Comment by Hedda** [comment]
> A CC2674P10 fw is now available; #545

@Koenkk Nice! Any chance you will consider also building and releasing ***Zigbee Router*** firmware for the CC2674 chips as well?

* https://github.com/Koenkk/Z-Stack-firmware/issues/538

**Comment by Hedda** [comment]
Now with the announcement of ”Suzi” - Sub-GHz Zigbee (and Zigbee 4.0) it would be awesome to have a CC1354P (CC1354P10) adapter!

* https://csa-iot.org/newsroom/the-connectivity-standards-alliance-announces-zigbee-4-0-and-suzi-empowering-the-next-generation-of-secure-interoperable-iot-devices/

  * https://csa-iot.org/all-solutions/suzi/

    * https://www.youtube.com/watch?v=eg3wZjp8uw8

  * "_**Alongside Zigbee 4.0, the Alliance introduces [Suzi](https://csa-iot.org/all-solutions/suzi/), the new brand for the standards-based wireless technology that extends the reach and reliability of IoT connectivity through long-range, Sub-GHz mesh networking. Built on the proven Zigbee network layer, Suzi combines long-range performance, low power consumption, and multi-vendor interoperability to unlock new opportunities in residential, commercial, and industrial applications. From connecting outdoor living spaces to enabling large-scale networks in buildings and cities, Suzi delivers robust, efficient communication in environments demanding extended coverage and minimal interference. Adhering to the same strong security principles that define all Alliance technologies, Suzi aligns with international standards to ensure a secure and trusted ecosystem. Its framework allows developers, manufacturers, and consumers the freedom to build and deploy interoperable devices from a global ecosystem of trusted suppliers. The Suzi Certification Program is planned to open in the first half of 2026, enabling manufacturers to begin certifying products that bring the benefits of long-range, low-power mesh networking to the connected world.**_"

FYI, Silicon Labs Zigbee EmberZnet SDK 7.2.0.0 was the first version that added support for Dual-PHY [Zigbee Smart Energy](https://github.com/zigpy/zigpy/issues/1709) needed to support Dual-Band on Silabs MG24 (EFR32MG24) if used in combination with a Si4468 Sub-GHz PA (Power Amplifier) in a Zigbee Coordinator. Zigbee Smart Energy Dual-Band 2.4GHz and Sub-GHz support for end-devices is also already available via some EFR32MG12 parts. 

* https://www.silabs.com/documents/public/user-guides/ug307-brd4155a-user-guide.pdf
  * https://www.silabs.com/wireless/proprietary/ezradiopro-sub-ghz-ics/device.si4468
    * https://www.silabs.com/documents/public/data-sheets/Si4468-7.pdf

<img width="1440" height="800" alt="image" src="https://github.com/user-attachments/assets/3accd9cc-9834-4940-bd16-9cf3bbf5b0f8" />

**Comment by Hedda** [comment]
FYI, @egony added support for cc2674p10 to his fork of cc2538-bsl:

* https://github.com/egony/cc2674p10_Ebyte/tree/main/firmware/flash_tools

smlight-cc-flasher also looks to have support for cc2674p now:

* https://github.com/smlight-tech/smlight-cc-flasher

Hardware example is SLZB-0xP10 which is based on CC2674p10: 

* https://smlight.tech/product/slzb-07p10

  * https://smlight.tech/flasher/#SLZB-07p10

**Comment by Hedda** [comment]
FYI, SMLIGHT has now also released additional combo adapters that feature CC2674P10 + some other radio chip as well:

SLZB-MR4U -> https://smlight.tech/global/slzbmr4

SLZB-MRW10U -> https://smlight.tech/global/slzbmrw10

**Comment by Hedda** [comment]
FYI, @Tarik2142 from [smlight-tech](https://github.com/smlight-tech) now also confirmed that their [smlight-cc-flasher](https://github.com/smlight-tech/smlight-cc-flasher) tool is compatible with CC2674 and CC1354 series:

* https://github.com/smlight-tech/smlight-cc-flasher

* https://smlight.tech/flasher/#SLZB-07p10

Quote from https://github.com/smlight-tech/smlight-cc-flasher/issues/10

> _**it work in our flasher tool https://smlight.tech/flasher/#SLZB-07p10 `--m33` option**_
